lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -93,8 +93,11 @@ def parse(document, options = {})
           attributes['id'] = primary_data['id'] if primary_data['id']
           relationships = primary_data['relationships'] || {}
 
-          filter_fields(attributes, options)
-          filter_fields(relationships, options)
+          attributes = transform_keys(attributes, options)
+          relationships = transform_keys(relationships, options)
```

Comment:
Given `payload = { data: { id: 1, type: user, attributes: { crazy-name: 'Lou' } } }` 
Then ` ActiveModelSerializers::Deserialization.jsonapi_parse(payload, only: [:crazy_name])` will make the following transforms

1. `{ data: { id: 1, type: user, attributes: { crazy_name: 'Lou' } } }` 
2. `{ id: 1, crazy_name: 'Lou' }`
3. `{crazy_name: 'Lou' }`